### PR TITLE
Create HOC and hooks to for jest mocks 

### PR DESCRIPTION
**Pull request by Nightrider0098**
## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Checklist before requesting a review
- [x] I have performed a self-review of my code
- [x] I have tested on Chrome and Firefox
- [ ] I have tested on a mobile device
- [ ] I have provided a screenshot or recording of changes in my PR if there were updates to the frontend

### CODE REVIEW

**Comment by ecurrencyhodler** [comment]
@Nightrider0098 can you tie this pr to the issue you are addressing?

**Comment by Nightrider0098** [comment]
@elraphty @kevkevinpal Can you please review the pr?

**Comment by Vayras** [comment]
@Nightrider0098 For the eslint fix keep the warnings under 72, so try giving type annotation where needed and remove unused dependencies

**Comment by Vayras** [comment]
@Nightrider0098 try rebasing to master for recent updates might fix your tests since the Test failing says "exceeded retry limit"

**Comment by Nightrider0098** [comment]
Should be fixed.

**Comment by Nightrider0098** [comment]
Passed all test case.

**Comment by Vayras** [comment]
@Nightrider0098 that's great all checks have passed @elraphty or @kevkevinpal will review this and get this merged.

**Comment by Nightrider0098** [comment]
@elraphty @kevkevinpal Can you please review my code?

File: src/store/util.ts
Comment (kevkevinpal) [inline src/store/util.ts:1]: I'm confused as to why we are creating a function in `./src/store/util.ts` and then only using it in a test function and nowhere else in the code base

Comment (ecurrencyhodler) [reply src/store/util.ts:1]: @Nightrider0098 can you address this comment?

Comment (Nightrider0098) [reply src/store/util.ts:1]: moved it to __tests__

**Comment by ecurrencyhodler** [comment]
Checks passed. Let's get another code review.

**Review by kevkevinpal — APPROVED** [review]
lgtm will merge after tests ci finishes running

**Pull request merged**